src/internal.c

```diff
@@ -17876,6 +17876,16 @@ int ReceiveData(WOLFSSL* ssl, byte* output, int sz, int peek)
                 goto startScr;
             }
         #endif
+    #ifdef WOLFSSL_TLS13
+        if (ssl->options.handShakeDone && ssl->curRL.type == handshake &&
+                                                                         peek) {
+            WOLFSSL_MSG("Got Handshake Messge in APP data");
+            if (ssl->buffers.inputBuffer.length == 0) {
+                ssl->error = WOLFSSL_ERROR_WANT_READ;
```

Comment:
Are you sure peek should short circuit the while loop here?  I always thought of PEEK like READ except that the data is left in the clearOutputBuffer.

Reply:
The data left in clearOutputBuffer is application data only. Handshake data is processed and removed.
It short-circuits when it processed handshake data but there is no application data available now.
So yes, I want it to short-circuit.